### ISSUE TITLE
Getting "Invalid (null or empty) build ID detected! Crash will be ignored by collector" when i have configured build_id

### ISSUE THREAD

**Issue by adame21** [newrelic/newrelic-react-native-agent#173]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
So I have an android + react native app, which our team added newrelic for monitoring
We seem to get these errors -

 E Agent.getBuildId() was unable to find a valid build Id. Crashes and handled exceptions will not be accepted.
 E Invalid (null or empty) build ID detected! Crash will be ignored by collector.
 E ExceptionHelper: java.util.UUID:fromString(UUID.java:194) RandomUUID[java.lang.IllegalArgumentException] Invalid UUID string: 
E Unable to get obfuscated flag in crash --------- beginning of crash 
E Invalid (null or empty) build ID detected! Crash will be ignored by collector.
 E Unable to get obfuscated flag in crash

## Steps to Reproduce
I just run the app and crash it  after 10 second delay to test this

  const appVersion = getVersion()
  NewRelicAgent.startAgent(config.NEW_RELIC_API_KEY, newRelicAgentConfiguration)
  NewRelicAgent.setMaxEventBufferTime(60)
  NewRelicAgent.setMaxEventPoolSize(4000)
  NewRelicAgent.setJSAppVersion(appVersion)
  NewRelicWrapper.updateSessionAttributes({env: config.ENV})
  setTimeout(() => {
    const test: any = {}
    console.log(test.should.fail)
  }, 10000)

I have in my build.gradle dependencies

        classpath "com.newrelic.agent.android:agent-gradle-plugin:7.0.0"

in the app build.gradle

        apply plugin: "newrelic"

## Expected Behavior

I expect the app to have the build ID provided in 
  NewRelicAgent.setJSAppVersion(appVersion)

## Relevant Logs / Console output
provided above

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible. )

* ex: Output of `npx react-native info`
* ex: Which platform is the issue happening on? (Android, iOS, tvOS, etc.)
* ex: Any conflicting packages installed?

## Additional context

[TIP]:  # ( Add any other context about the problem here. )

ex: If appropriate and possible, it would be helpful to add [Charles traces](https://forum.newrelic.com/s/hubtopic/aAX8W0000008bQlWAI/relic-solution-setting-up-and-collecting-charles-trace-for-android-apps).

**Comment by ndesai-newrelic** [comment]
@adame21 can you share your Gradle version and AGP version for this? you need to use Gradle 7.4.or higher and same for AGP also.

**Comment by adame21** [comment]
Hi thanks for the reply
currently we have
classpath "com.android.tools.build:gradle:7.0.4"
classpath "com.newrelic.agent.android:agent-gradle-plugin:7.0.0"

which means we need to upgrade versions and test.
unfortunately its not a trivial thing in our project, so ill have to take some time to do it.
ill update if that resolved it here once thats done
for now should i close the issue until i can update ?
thanks !

**Comment by ShalevC** [comment]
@ndesai-newrelic there is no reference in the docs about gradle 7.4.+ .
are you sure about that ?

**Comment by ndesai-newrelic** [comment]
@ShalevC yes,it is related AGP 7 changes, please check our androud agent gradle requirments.

**Comment by MeetSwiggy** [comment]
@ndesai-newrelic Even I am facing similar issue. Crashes are not getting reported in Newrelic post upgrade to v7.3.0.
My gradle version is 7.5.1 and AGP is 7.2.1. 

I cannot find anything related to gradle update in the [release notes](https://docs.newrelic.com/docs/release-notes/mobile-release-notes/android-release-notes/android-730/).

Can you point me to the right doc where the upgrade of gradle plugin is mentioned?

**Comment by ndesai-newrelic** [comment]
@MeetSwiggy you need to use AGP version 7.4.0 or higher with gradle version 7.5.1
